src/utils/helpers/application/buildQueryFilters.ts

```diff
@@ -0,0 +1,119 @@
+import { DateTime } from 'luxon'
+import { QueryObject } from '../../hooks/useRouter'
+import { ApplicationStatus, ApplicationOutcome } from '../../../utils/generated/graphql'
+
+export default function buildQueryFilters(filters: QueryObject) {
+  const graphQLfilter = Object.entries(filters).reduce((filterObj, [key, value]) => {
+    if (!mapQueryToFilterField?.[key]) return filterObj
+    return { ...filterObj, ...mapQueryToFilterField[key](value) }
+  }, {})
+  return graphQLfilter
+}
+interface FilterMap {
+  [key: string]: (value: string) => object
+}
+
+const mapQueryToFilterField: FilterMap = {
+  type: (value: string) => {
+    return { templateCode: { equalToInsensitive: value } }
+  },
+  // category -- not yet implemented in schema
+  stage: (values: string) => {
+    return {
+      stage: inList(values),
+    }
+  },
+  status: (values: string) => {
+    return { status: inEnumList(values, ApplicationStatus) }
+  },
+  outcome: (values: string) => {
+    return { outcome: inEnumList(values, ApplicationOutcome) }
+  },
+  // action
+  // assigned
+  // consolidator
+  applicant: (values: string) => {
+    return {
+      or: [
+        { applicant: inList(values) },
+        { applicantFirstName: inList(values) },
+        { applicantLastName: inList(values) },
+        { applicantUsername: inList(values) },
+      ],
+    }
+  },
+  org: (values: string) => {
+    return { orgName: inList(values) }
+  },
+  lastActiveDate: (value: string) => {
+    const [startDate, endDate] = parseDateString(value)
+    console.log('Dates:', startDate, endDate)
+    return { lastActiveDate: { greaterThanOrEqualTo: startDate, lessThan: endDate } }
+  },
+  // deadlineDate (TBD)
+  search: (value: string) => {
+    return {
+      or: [
+        { name: { includesInsensitive: value } },
+        { applicantUsername: { includesInsensitive: value } },
+        { applicant: { includesInsensitive: value } },
+        { orgName: { includesInsensitive: value } },
+        { templateName: { includesInsensitive: value } },
+        { stage: { startsWithInsensitive: value } },
+      ],
+    }
+  },
```

Comment:
Ah yes, much nicer. I never knew how to use the shorthand return form when it was just returning an object, because it always interpreted the outer {} as the function wrapper. Now I see you can just put () around it! Very nice, thanks.